test/profiler/test_profiler_tree.py

```diff
@@ -538,87 +538,95 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
                   aten::empty
                   aten::fill_
               nn.Module: MyModule_0
-                <built-in method _get_tracing_state of PyCapsule object at 0xXXXXXXXXXXXX>
-                test_profiler_tree.py(...): forward
-                  nn.Module: ReLU_0
-                    <built-in method _get_tracing_state of PyCapsule object at 0xXXXXXXXXXXXX>
-                    torch/nn/modules/activation.py(...): forward
-                      torch/nn/functional.py(...): relu
-                        <built-in function _has_torch_function_unary>
-                        <built-in method relu of type object at 0xXXXXXXXXXXXX>
-                          aten::relu
-                            aten::clamp_min
-                  nn.Module: Linear_0
-                    <built-in method _get_tracing_state of PyCapsule object at 0xXXXXXXXXXXXX>
-                    torch/nn/modules/linear.py(...): forward
-                      torch/nn/modules/module.py(...): __getattr__
-                      torch/nn/modules/module.py(...): __getattr__
-                      <built-in function linear>
-                        aten::linear
-                          aten::t
-                            aten::transpose
-                              aten::as_strided
-                          aten::matmul
-                            aten::unsqueeze
-                              aten::as_strided
-                            aten::mm
-                              aten::resolve_conj
-                              aten::resolve_conj
-                              aten::resolve_conj
-                            aten::squeeze_
-                              aten::as_strided_
-                          aten::add_
-                  nn.Module: ReLU_1
-                    <built-in method _get_tracing_state of PyCapsule object at 0xXXXXXXXXXXXX>
-                    torch/nn/modules/activation.py(...): forward
-                      torch/nn/functional.py(...): relu
-                        <built-in function _has_torch_function_unary>
-                        <built-in method relu of type object at 0xXXXXXXXXXXXX>
-                          aten::relu
-                            aten::clamp_min
+                torch/nn/modules/module.py(...): _call_impl
```

Comment:
Kinda strange too that after I made the changes here and got it to pass locally it's failing remotely 😕

Reply:
@davidberard98 as well

Reply:
also cc @aaronenyeshi @chaekit in case, they probably have more context on this than me

Reply:
Can we preserve the previous behavior? This looks like it will add another layer for every nn.module, making the python tracing stacks harder to read and deeper.

Reply:
@aaronenyeshi it won't be possible for me to land this PR and preserve the old behavior. In this case, we believe the change is worth it primarily because users have been tripped up with not being able to save compiled modules

Reply:
@aaronenyeshi are there any trick we can do in our impl/in the dispatcher to hide this new indirection from the trace?